src/Raven.Server/Documents/ShardedTcpHandlers/ShardedSubscriptionConnection.cs

```diff
@@ -18,17 +18,49 @@
 using Raven.Server.ServerWide;
 using Sparrow.Json;
 using Sparrow.Json.Parsing;
+using Sparrow.Server;
+using Sparrow.Utils;
 
 namespace Raven.Server.Documents.ShardedTcpHandlers
 {
     public class ShardedSubscriptionConnection : SubscriptionConnectionBase, IDisposable
     {
         private readonly Dictionary<string, SubscriptionShardHolder> _shardWorkers = new Dictionary<string, SubscriptionShardHolder>();
         public static ConcurrentDictionary<string, ShardedSubscriptionConnection> Connections = new ConcurrentDictionary<string, ShardedSubscriptionConnection>();
+        public  AsyncManualResetEvent _mre;
 
         private ShardedSubscriptionConnection(ServerStore serverStore, TcpConnectionOptions tcpConnection, IDisposable tcpConnectionDisposable, JsonOperationContext.MemoryBuffer buffer)
             : base(tcpConnection, serverStore, buffer, tcpConnectionDisposable, tcpConnection.ShardedContext.DatabaseName, CancellationTokenSource.CreateLinkedTokenSource(tcpConnection.ShardedContext.DatabaseShutdown))
         {
+            _mre = new AsyncManualResetEvent();
+        }
+
+       /*
+        * The subscription worker (client) connects to an orchestrator (SendShardedSubscriptionDocuments). 
+        * The orchestrator initializes shard subscription workers (StartShardSubscriptionWorkers).
+        * ShardedSubscriptionWorkers are maintaining the connection with subscription on each shard.
+        * The orchestrator maintains the connection with the client and checks if there is an available batch on each Sharded Worker (MaintainConnectionWithClientWorker).
+        * Handle batch flow:
+        * Orchestrator sends the batch to the client (WriteBatchToClientAndAck).
```

Comment:
Where are we handling `WhoseTaskIsIt` for the orchestrator node?

Reply:
Why do we need `WhoseTaskIsIt` on orchestrator node?
Orchestrator only redirects the data so it can reside on node that even don't have the data it self.

I guess we can optimize it by explicitly setting the orchestrator node to be the same node as one of the shards (so we can save some traffic/latency) ?